### PR TITLE
[SW] New Benchmark: FFT (+plot performance)

### PR DESCRIPTION
**Pull request by mp-17**
Add FFT to apps.

This FFT version exploits Ara's VRF to buffer all the samples.

### Fixed

- Fix VALU issue_counter initialization for mask logical operations

### Added

- Add FFT benchmark and print its performance

### Changed

- Replace `apps/common/script/datagen.sh` with new input data source-of-truth (`apps/common/default_arguments.mk`) during app compilation

## Checklist

- [x] Automated tests pass
- [x] Changelog updated
- [x] Code style guideline is observed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -187,7 +187,7 @@ jobs:
     - name: Untar LLVM
       run: tar xvf tc-llvm.tar
     - name: Compile applications
-      run: config=${{ matrix.ara_config }} make -C apps
+      run: config=${{ matrix.ara_config }} make -C apps ENV_DEFINES="-DFFT_SAMPLES=64"
```
Comment (suehtamacv) [inline .github/workflows/ci.yml:190]: Can we just set this to be default?

**Review by suehtamacv — CHANGES_REQUESTED** [review]

File: apps/fft/script/gen_data.py~ [hunk 2]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (mp-17) [reply apps/fft/script/gen_data.py~:1]: Thanks, did not see it

File: apps/fft/support_data.h [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Author: Matteo Perotti <mperotti@iis.ee.ethz.ch>
+
+// #include <stdint.h>
```
[the remark this answers is not in the document]
Comment (mp-17) [reply apps/fft/support_data.h:19]: I will remove it, it was a dead statement :dagger:

**Comment by mp-17** [comment]
I added a single source of truth for the default app parameters :-)

**Pull request merged**